### PR TITLE
[filebeat] Fix long filepaths in diagnostics exceeding max path limits on Windows

### PR DESCRIPTION
**Pull request by aleksmaus**
## Proposed commit message

Fix long filepaths in diagnostics exceeding max path limits on Windows.

Following the first proposal here
https://github.com/elastic/elastic-agent/issues/3758#issuecomment-1830229924
Added ```IDWithoutName``` that doesn't include the source name and using that instead.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes https://github.com/elastic/elastic-agent/issues/3758

## Screenshots

The screenshot that shows the diagnostics collected before and after this change. The earlier diagnostics shows the long path that includes the sanitized okta log collection URL. The diagnostics after this change have shorter path that doesn't include the sanitized URL in the path.
Screenshot 2024-09-19 at 2 36 12 PM

### CODE REVIEW

**Comment by aleksmaus** [comment]
This is odd linter complain to the line:
https://github.com/elastic/beats/blob/43d80af2aea60b0c45711475d114e118d90c4581/x-pack/filebeat/input/cel/input.go#L1253
```
  Error: x-pack/filebeat/input/cel/input.go:1253:7: rest declared and not used (typecheck)
  	key, rest, more := strings.Cut(path, ".")
```

Can't reproduce this on my machine and the ```rest``` variable is clearly used later in this function

**Comment by efd6** [comment]
@aleksmaus [This](https://github.com/elastic/beats/pull/40909#issuecomment-2361974985) is #39589.

File: x-pack/filebeat/input/cel/input.go
Comment (efd6) [inline x-pack/filebeat/input/cel/input.go:1253]: Please remove this.

File: x-pack/filebeat/input/cel/input_test.go
Comment (efd6) [inline x-pack/filebeat/input/cel/input_test.go:1]: Please revert these linter apologies. The job of the linter is to improve the quality of the code, these additions do not do that.

File: x-pack/filebeat/input/httpjson/input_test.go
Comment (efd6) [inline x-pack/filebeat/input/httpjson/input_test.go:1362]: Please revert this.

Comment (aleksmaus) [reply x-pack/filebeat/input/httpjson/input_test.go:1362]: The old code doesn't pass the linter. This doesn't change the existing behavior of the unit test.

File: x-pack/filebeat/input/cel/input.go
Comment (aleksmaus) [reply x-pack/filebeat/input/cel/input.go:1253]: I disagree. This been used before in this repo in the other place.
The linter in CI flags this and this is false positive, please look at the code and the linter error above.
This is also not a new code and not related to my change.
The linter on my local machine doesn't fail here as well, so I can't reproduce the issue.

**Comment by aleksmaus** [comment]
> @aleksmaus [This](https://github.com/elastic/beats/pull/40909#issuecomment-2361974985) is #39589.

So what is the course of action in order to get this change in, which unrelated to all the linter complains that surfaced just because I had to touch these files?

File: x-pack/filebeat/input/httpjson/input_test.go
Comment (efd6) [reply x-pack/filebeat/input/httpjson/input_test.go:1362]: The linter does not block merging.

File: x-pack/filebeat/input/cel/input.go
Comment (efd6) [reply x-pack/filebeat/input/cel/input.go:1253]: The linter directive here is being used to work around a bug. This is described in the issue I linked above.

**Comment by efd6** [comment]
The action is no action. Linter happiness is not a requirement for merging.

File: x-pack/filebeat/input/entityanalytics/provider/okta/okta.go
Comment (efd6) [inline x-pack/filebeat/input/entityanalytics/provider/okta/okta.go:1]: Please revert this.

**Review by efd6 — CHANGES_REQUESTED** [review]

**Comment by aleksmaus** [comment]
> The action is no action. Linter happiness is not a requirement for merging.

Thank you for clarification. I didn't know. It is different in different repositories. Some of the repos are blocking PR on linter errors. Will revert. Let the linter complain away then.

File: x-pack/filebeat/input/entityanalytics/provider/okta/okta.go
Comment (aleksmaus) [reply x-pack/filebeat/input/entityanalytics/provider/okta/okta.go:1]: This is actually a correct way to handle errors in Go. Why revert?

Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/okta/okta.go:1]: It's not necessary in this case. We have complete control over the code path that returns this error and it never wraps `io.EOF`. From our code, we don't and make sure of that. The stdlib part doesn't and is constrained by go1compat not to change that behaviour.

**Review by efd6 — APPROVED** [review]
Thanks, this is good to have in.

**Review by andrewkroh — APPROVED** [review]

**Comment by aleksmaus** [comment]
@ycombinator @belimawr @AndersonQ Guys, I need 👍  on this one from one of the code owners team members. Pretty please? :)

**Comment by belimawr** [comment]
> @ycombinator @belimawr @AndersonQ Guys, I need 👍 on this one from one of the code owners team members. Pretty please? :)

I'm looking at it right now ;)

**Review by belimawr — COMMENTED** [review]
LGTM, but I have two small comments:
1. Is the goal of this PR to ensure the max path limit is never going to be exceeded or is it to just reduce the input name used by some inputs?
2. Unless it's a 9.0 breaking change, all PRs should be backported to 8.x. I've added the backport label.

**Comment by aleksmaus** [comment]
> 1. Is the goal of this PR to ensure the max path limit is never going to be exceeded or is it to just reduce the input name used by some inputs?

According the original issue, the goal was to reduce the pathing length. The other path parts are more or less have predictable length.

**Comment by aleksmaus** [comment]
> 2\. Unless it's a 9.0 breaking change, all PRs should be backported to 8.x. I've added the backport label.

Good to know. Thanks!

**Review by belimawr — APPROVED** [review]

**Review by mauri870 — APPROVED** [review]

**Comment by AndersonQ** [comment]
I don't wanna block, but I don't see any tests to ensure the right name is used on the files. Unless it'd be too complicated, could you add some?

**Pull request merged**
